src/registry.jl

```diff
@@ -60,6 +59,9 @@ end
 
 detect_compressed(io, len=getlength(io); kwargs...) = detect_compressor(io, len; kwargs...) !== nothing
 
+const compressed_fits_exten = r"\.(fit|fits|fts|FIT|FITS|FTS)\.(gz|GZ)\>"
+name_matches_compressed_fits(io) = (:name ∈ propertynames(io)) && endswith(io.name, compressed_fits_exten)
```

Comment:
Curiously, what if `propertynames` doesn't contain `:name`? Will this break the functionality of `detect_rdata`?

How about

```julia
name_matches_compressed_fits(io::IOStream) = endswith(io.name, compressed_fits_exten)
name_matches_compressed_fits(io::IO) = false # does FITSIO and AstroImages support generic IO object (e.g., `IOBuffer`)?
```

Reply:
Looking at the logic of this again, I think it's okay. 
If we detect that it's compressed, then if there's no name or the name does not match a compressed fits file, it falls back to the compressed R data path.

Ideally someone with compressed R data could remove the file extension and test this path out.